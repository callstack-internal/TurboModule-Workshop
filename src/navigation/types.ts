import { PerroConEstado } from '../types/dog.ts';

export type RouteNamesParamList = RootStackParamList | DrawerParamList;

export type RootStackParamList = {
  Home: undefined;
  AdoptedDogs: undefined;
  DogDetails: { dog: PerroConEstado };
  HowItWorks: undefined;
};

export type DrawerParamList = {
  Main: undefined;
  HowItWorks: undefined;
};
