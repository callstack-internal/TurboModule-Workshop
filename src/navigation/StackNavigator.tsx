import React from 'react';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import TabNavigator from './TabNavigator';
import DogDetailsScreen from '../screens/DogDetailsScreen';
import HowItWorksScreen from '../screens/HowItWorksScreen';
import { RootStackParamList } from '../screens/DogDetailsScreen';

// Create the stack navigator
const Stack = createNativeStackNavigator<RootStackParamList>();

function StackNavigator() {
  return (
    <Stack.Navigator
      screenOptions={{
        headerStyle: {
          backgroundColor: '#ffffff',
        },
        headerTitleStyle: {
          fontWeight: 'bold',
          color: '#212529',
        },
        headerTintColor: '#28a745',
      }}
    >
      <Stack.Screen name="Home" component={TabNavigator} options={{ headerShown: false }} />
      <Stack.Screen
        name="DogDetails"
        component={DogDetailsScreen}
        options={({ route }) => {
          console.log('[DEBUG] DogDetails route params:', route.params);
          // Add error handling for the case where route.params or route.params.dog is undefined
          return {
            title: route.params?.dog?.nombre || 'Detalles del perro',
          };
        }}
      />
      <Stack.Screen
        name="HowItWorks"
        component={HowItWorksScreen}
        options={{ title: 'CÃ³mo funciona' }}
      />
    </Stack.Navigator>
  );
}

export default StackNavigator;
