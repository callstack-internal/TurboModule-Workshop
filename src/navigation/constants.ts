export const Screens = {
  Main: {
    title: 'Catálogo de Perros',
    name: 'Main',
  },
  HowItWorks: {
    title: 'Cómo funciona',
    name: 'HowItWorks',
  },
  Home: {
    title: 'Home',
    name: 'Home',
  },
  DogDetails: {
    title: 'Detalles',
    name: 'DogDetails',
  },
  Adoptar: {
    title: 'Adoptar',
    name: 'Adoptar',
  },
  MisPerros: {
    title: 'Mis Perros',
    name: 'MisPerros',
  },
} as const;
