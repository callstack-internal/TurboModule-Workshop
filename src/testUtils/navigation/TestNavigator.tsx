import React, { useEffect } from 'react';

import {
  createNavigatorFactory,
  useNavigationBuilder,
  StackRouter,
  TabRouter,
  DrawerRouter,
  DefaultRouterOptions,
  StackRouterOptions,
  TabRouterOptions,
  DrawerRouterOptions,
  NavigationHelpers,
  ParamListBase,
  RouterFactory,
  Router,
  NavigationState,
  NavigationAction,
  NavigationContainer,
} from '@react-navigation/native';
import type {
  StackNavigationState,
  TabNavigationState,
  DrawerNavigationState,
} from '@react-navigation/native';
import type {
  StackActionHelpers,
  TabActionHelpers,
  DrawerActionHelpers,
} from '@react-navigation/native';
import { Text } from 'react-native';

type TestScreenOptions = Record<string, unknown>;

type NavigatorType = 'stack' | 'tabs' | 'drawer';

interface TestNavigatorImplProps extends DefaultRouterOptions {
  type: NavigatorType;
  screenOptions?: TestScreenOptions;
  initialRouteName?: string;
  children: React.ReactNode;
}

type AnyNavigationState =
  | StackNavigationState<ParamListBase>
  | TabNavigationState<ParamListBase>
  | DrawerNavigationState<ParamListBase>;

type AnyRouterOptions = StackRouterOptions | TabRouterOptions | DrawerRouterOptions;

type AnyActionHelpers =
  | StackActionHelpers<ParamListBase>
  | TabActionHelpers<ParamListBase>
  | DrawerActionHelpers<ParamListBase>;

type ExtractedAction = Exclude<
  Parameters<Router<NavigationState, NavigationAction>['getStateForAction']>[0],
  undefined
>;

/**
 * Custom navigator component used for testing React Navigation screens.
 * Supports `stack`, `tabs`, and `drawer` types.
 * Renders all screens at once and injects a `focused` prop based on navigation state.
 *
 * Useful in tests to simulate navigation behavior with a lightweight navigator.
 */
const TestNavigatorImpl = ({ type, ...rest }: TestNavigatorImplProps): React.ReactElement => {
  const router = (
    type === 'drawer' ? DrawerRouter : type === 'stack' ? StackRouter : TabRouter
  ) as RouterFactory<AnyNavigationState, ExtractedAction, AnyRouterOptions>;

  const {
    state,
    descriptors,
    NavigationContent: Content,
  } = useNavigationBuilder<
    AnyNavigationState,
    AnyRouterOptions,
    AnyActionHelpers,
    TestScreenOptions,
    NavigationHelpers<ParamListBase>
  >(router, rest);

  return <Content>{state.routes.map((route) => descriptors[route.key]?.render())}</Content>;
};

/**
 * Creates a navigator factory for the TestNavigator.
 * This wraps the custom `TestNavigator` using `createNavigatorFactory`,
 * enabling use of `Navigator` and `Screen` components just like built-in navigators.
 *
 * Use this when integrating `TestNavigator` into `NavigationContainer`.
 */
const createTestNavigatorFactory = createNavigatorFactory(TestNavigatorImpl);

type LayoutProps = {
  name: string;
  focused: boolean;
  navigation: NavigationHelpers<ParamListBase>;
  children: React.ReactNode;
};

const CustomLayout = ({ name, focused, navigation, children }: LayoutProps): React.ReactElement => {
  useEffect(() => {
    const state = navigation.getState();
    const currentRoute = state.routes[state.index];

    if (focused && currentRoute?.name !== name) {
      navigation.navigate(name);
    } else if (!focused && currentRoute?.name !== 'Dummy') {
      navigation.navigate('Dummy');
    }
  }, [focused, name, navigation]);

  return <>{children}</>;
};

const TestLayout = ({ name, focused, ...props }: LayoutProps): React.ReactElement => {
  return <CustomLayout {...props} name={name} focused={focused} />;
};

export const TestNavigator = ({
  children,
  type = 'stack',
  name = 'TestScreen',
  focused = true,
  screenParams,
  dummyName = 'Dummy',
}: {
  children: React.ReactNode;
  type?: NavigatorType;
  name?: string;
  focused?: boolean;
  screenParams?: object;
  dummyName?: string;
}) => {
  const Test = createTestNavigatorFactory();

  return (
    <NavigationContainer>
      <Test.Navigator
        key={type}
        type={type}
        initialRouteName={focused ? name : dummyName}
        layout={
          /* eslint-disable-next-line react/no-unstable-nested-components */
          (props: Omit<LayoutProps, 'name' | 'focused'>) => (
            <TestLayout {...props} name={name} focused={focused} />
          )
        }
      >
        <Test.Screen name={name} getId={() => name} initialParams={screenParams}>
          {() => children}
        </Test.Screen>
        <Test.Screen name={dummyName} getId={() => dummyName}>
          {() => <Text testID="dummy-screen">Dummy</Text>}
        </Test.Screen>
      </Test.Navigator>
    </NavigationContainer>
  );
};
