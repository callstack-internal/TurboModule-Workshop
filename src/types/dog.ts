export interface Perro {
  nombre: string;
  raza: string;
  foto: string;
  edad?: number;
}

export interface Chip {
  chipId: string;
  fechaRegistro: string;
}

export interface NoAdoptable {
  motivo: 'enfermo' | 'muy joven';
  fechaDisponible: string;
}

export type PerroConChip = Perro & Chip;

export type PerroConEstado = Perro | (Perro & NoAdoptable);