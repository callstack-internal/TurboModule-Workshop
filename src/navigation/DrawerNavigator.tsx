import React from 'react';
import {
  createDrawerNavigator,
  DrawerContentScrollView,
  DrawerItemList,
} from '@react-navigation/drawer';
import { View, Text, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import StackNavigator from './StackNavigator';
import HowItWorksScreen from '../screens/HowItWorksScreen';

// Define the drawer navigator param list
export type DrawerParamList = {
  Main: undefined;
  HowItWorks: undefined;
};

const Drawer = createDrawerNavigator<DrawerParamList>();

// Custom drawer content component
const CustomDrawerContent = (props: any) => (
  <DrawerContentScrollView {...props}>
    <View style={styles.drawerHeader}>
      <Text style={styles.drawerTitle}>Adopci칩n de Perros</Text>
      <Text style={styles.drawerSubtitle}>Encuentra tu compa침ero ideal</Text>
    </View>
    <DrawerItemList {...props} />
  </DrawerContentScrollView>
);

function DogIcon({ color, size }: { color: string; size: number }) {
  return <Icon name="dog" size={size} color={color} style={styles.icon} />;
}

function HelpIcon({ color, size }: { color: string; size: number }) {
  return <Icon name="help-circle" size={size} color={color} style={styles.icon} />;
}

function DrawerNavigator() {
  return (
    <Drawer.Navigator
      screenOptions={{
        headerShown: false,
        drawerActiveTintColor: '#28a745',
        drawerInactiveTintColor: '#212529',
        drawerLabelStyle: {
          marginLeft: -16,
          fontSize: 16,
        },
      }}
      drawerContent={CustomDrawerContent}
    >
      <Drawer.Screen
        name="Main"
        component={StackNavigator}
        options={{
          title: 'Cat치logo de Perros',
          drawerIcon: DogIcon,
        }}
      />
      <Drawer.Screen
        name="HowItWorks"
        component={HowItWorksScreen}
        options={{
          title: 'C칩mo funciona',
          drawerIcon: HelpIcon,
        }}
      />
    </Drawer.Navigator>
  );
}

const styles = StyleSheet.create({
  drawerHeader: {
    padding: 16,
    backgroundColor: '#f8f9fa',
    borderBottomWidth: 1,
    borderBottomColor: '#e9ecef',
  },
  drawerTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#212529',
    marginBottom: 4,
  },
  drawerSubtitle: {
    fontSize: 14,
    color: '#6c757d',
  },
  icon: {
    marginRight: 10,
  },
});

export default DrawerNavigator;
