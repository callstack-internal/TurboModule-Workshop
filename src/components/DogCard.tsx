import React, { useMemo } from 'react';
import {
  View,
  Text,
  Image,
  StyleSheet,
  TouchableOpacity
} from 'react-native';
import { PerroConEstado } from '../types/dog';
import { mostrarInfoPerro, esAdoptable } from '../utils/dogUtils';

interface DogCardProps {
  perro: PerroConEstado;
}

/**
 * DogCard component displays information about a single dog
 */
const DogCard: React.FC<DogCardProps> = React.memo(({ perro }) => {
  // Get dog information and adoption status
  const info = useMemo(() => mostrarInfoPerro(perro), [perro]);
  const adoptable = useMemo(() => esAdoptable(perro), [perro]);

  // Get dog age display text
  const ageText = useMemo(() => {
    if (perro.edad === undefined) {
      return 'Edad desconocida';
    }
    return perro.edad === 1 ? '1 año' : `${perro.edad} años`;
  }, [perro.edad]);

  return (
    <TouchableOpacity
      style={styles.cardContainer}
      activeOpacity={0.9}
    >
      <View style={styles.card}>
        <Image
          source={{ uri: perro.foto }}
          style={styles.image}
          resizeMode="cover"
        />

        <View style={styles.infoContainer}>
          <Text style={styles.name}>{perro.nombre}</Text>
          <Text style={styles.breed}>{perro.raza}</Text>
          <Text style={styles.age}>{ageText}</Text>

          <View style={styles.divider} />

          <View style={styles.statusContainer}>
            <Text style={styles.infoText}>{info}</Text>
            <View style={[
              styles.statusBadge,
              adoptable ? styles.adoptableBadge : styles.notAdoptableBadge
            ]}>
              <Text style={styles.statusText}>
                {adoptable ? 'Disponible para adopción' : 'No adoptable aún'}
              </Text>
            </View>

            {!adoptable && 'motivo' in perro && (
              <Text style={styles.reasonText}>
                Motivo: {perro.motivo === 'enfermo' ? 'Está enfermo' : 'Es muy joven'}
              </Text>
            )}
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );
});

// No need for Dimensions since we're using relative sizing

const styles = StyleSheet.create({
  cardContainer: {
    marginBottom: 16,
    borderRadius: 12,
    backgroundColor: '#ffffff',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
    overflow: 'hidden',
  },
  card: {
    flexDirection: 'column',
    borderRadius: 12,
  },
  image: {
    width: '100%',
    height: 200,
    borderTopLeftRadius: 12,
    borderTopRightRadius: 12,
  },
  infoContainer: {
    padding: 16,
  },
  name: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#212529',
    marginBottom: 4,
  },
  breed: {
    fontSize: 16,
    color: '#495057',
    marginBottom: 4,
    textTransform: 'capitalize',
  },
  age: {
    fontSize: 14,
    color: '#6c757d',
    marginBottom: 8,
  },
  divider: {
    height: 1,
    backgroundColor: '#e9ecef',
    marginVertical: 12,
  },
  statusContainer: {
    marginTop: 4,
  },
  infoText: {
    fontSize: 15,
    color: '#495057',
    marginBottom: 12,
  },
  statusBadge: {
    paddingVertical: 6,
    paddingHorizontal: 12,
    borderRadius: 20,
    alignSelf: 'flex-start',
    marginBottom: 8,
  },
  adoptableBadge: {
    backgroundColor: '#d4edda',
  },
  notAdoptableBadge: {
    backgroundColor: '#f8d7da',
  },
  statusText: {
    fontSize: 14,
    fontWeight: '500',
    color: '#212529',
  },
  reasonText: {
    fontSize: 14,
    color: '#dc3545',
    marginTop: 4,
    fontStyle: 'italic',
  },
});

export default DogCard;
