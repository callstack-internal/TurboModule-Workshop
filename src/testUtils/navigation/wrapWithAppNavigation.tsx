import React from 'react';

import {
  InitialState,
  NavigationContainer,
  NavigationContainerRef,
} from '@react-navigation/native';
import DrawerNavigator from '../../navigation/navigators/DrawerNavigator.tsx';

type WrapWithAppNavigationProps = {
  initialState?: InitialState;
  navigationRef?: React.RefObject<NavigationContainerRef<any>>;
};

/**
 * Wraps the full app navigator (`DrawerNavigator`) in a `NavigationContainer` for testing.
 *
 * This is the preferred utility when writing tests for **real screen components or components that navigate** that are part of
 * the actual app navigation tree. It gives you a full, realistic navigation environment without mocks,
 * including nested navigators, screen options, transitions, etc.
 *
 * Unlike `TestNavigator`, this uses your actual app's navigator configuration, making it ideal for testing:
 * - Navigation behavior (e.g., `navigate`, `goBack`, deep linking)
 * - Screen lifecycle hooks (e.g., `useFocusEffect`, `useNavigation`)
 * - Integration-level rendering of screen flows
 *
 * ⚠️ Note: This should be used when testing **screen components**.
 * If you're only testing a non-screen component (e.g. custom hook, modal, header)
 * that requires navigation context, consider using `TestNavigator` or `createTestNavigator` instead for a minimal setup.
 *
 */
export const wrapWithAppNavigation = ({
  initialState,
  navigationRef,
}: WrapWithAppNavigationProps) => {
  return (
    <NavigationContainer initialState={initialState} ref={navigationRef}>
      <DrawerNavigator />
    </NavigationContainer>
  );
};
