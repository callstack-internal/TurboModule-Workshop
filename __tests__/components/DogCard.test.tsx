import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import DogCard from '../../src/components/DogCard';
import { PerroConEstado } from '../../src/types/dog';
import { TestNavigator } from '../../src/testUtils/navigation/TestNavigator';

describe('DogCard', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const adoptableDog: PerroConEstado = {
    nombre: 'Max',
    raza: 'labrador',
    foto: 'https://example.com/dog.jpg',
    edad: 3,
    distancia: 2.5,
  };

  const nonAdoptableDog: PerroConEstado = {
    nombre: 'Luna',
    raza: 'golden retriever',
    foto: 'https://example.com/dog2.jpg',
    edad: 1,
    motivo: 'muy joven',
    fechaDisponible: '2023-12-31',
  };

  const nearbyDog: PerroConEstado = {
    nombre: 'Rocky',
    raza: 'bulldog',
    foto: 'https://example.com/dog3.jpg',
    edad: 5,
    distancia: 0.5,
  };

  const unknownAgeDog: PerroConEstado = {
    nombre: 'Bella',
    raza: 'poodle',
    foto: 'https://example.com/dog4.jpg',
  };

  const renderDogCard = (perro: PerroConEstado) =>
    render(
      <TestNavigator name="DogCard" dummyName="DogDetails">
        <DogCard perro={perro} />
      </TestNavigator>,
    );

  test('renders adoptable dog correctly', () => {
    const { getByText, queryByText } = renderDogCard(adoptableDog);

    expect(getByText('Max')).toBeTruthy();
    expect(getByText('labrador')).toBeTruthy();
    expect(getByText('3 años')).toBeTruthy();
    expect(getByText('A 2.5 km de tu ubicación')).toBeTruthy();
    expect(getByText('Disponible para adopción')).toBeTruthy();
    expect(getByText('Max es de raza labrador')).toBeTruthy();
    expect(queryByText(/Motivo:/)).toBeNull();
  });

  test('renders non-adoptable dog correctly', () => {
    const { getByText } = renderDogCard(nonAdoptableDog);

    expect(getByText('Luna')).toBeTruthy();
    expect(getByText('golden retriever')).toBeTruthy();
    expect(getByText('1 año')).toBeTruthy();
    expect(getByText('No adoptable aún')).toBeTruthy();
    expect(getByText('Luna es de raza golden retriever')).toBeTruthy();
    expect(getByText('Motivo: Es muy joven')).toBeTruthy();
  });

  test('renders nearby dog with nearby badge', () => {
    const { getByText } = renderDogCard(nearbyDog);

    expect(getByText('A 0.5 km de tu ubicación')).toBeTruthy();
    expect(getByText('¡Cercano a ti!')).toBeTruthy();
  });

  test('renders dog with unknown age correctly', () => {
    const { getByText } = renderDogCard(unknownAgeDog);

    expect(getByText('Edad desconocida')).toBeTruthy();
  });

  test('navigates to dog details when pressed', async () => {
    const { getByText, findByTestId } = renderDogCard(adoptableDog);

    fireEvent.press(getByText('Max'));

    // Confirm navigation to Dummy (DogDetails) screen
    expect(await findByTestId('dummy-screen')).toBeTruthy();
  });

  test('matches snapshot', () => {
    const { toJSON } = renderDogCard(adoptableDog);

    expect(toJSON()).toMatchSnapshot();
  });
});
