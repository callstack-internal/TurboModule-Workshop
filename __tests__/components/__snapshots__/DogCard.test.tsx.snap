// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogCard matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "elevation": 3,
      "marginBottom": 16,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      {
        "borderRadius": 12,
        "flexDirection": "column",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "uri": "https://example.com/dog.jpg",
        }
      }
      style={
        {
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "height": 200,
          "width": "100%",
        }
      }
    />
    <View
      style={
        {
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#212529",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Max
      </Text>
      <Text
        style={
          {
            "color": "#495057",
            "fontSize": 16,
            "marginBottom": 4,
            "textTransform": "capitalize",
          }
        }
      >
        labrador
      </Text>
      <Text
        style={
          {
            "color": "#6c757d",
            "fontSize": 14,
            "marginBottom": 8,
          }
        }
      >
        3 años
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginTop": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#0066cc",
              "fontSize": 14,
              "fontWeight": "500",
            }
          }
        >
          A 
          2.5
           km de tu ubicación
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#e9ecef",
            "height": 1,
            "marginVertical": 12,
          }
        }
      />
      <View
        style={
          {
            "marginTop": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#495057",
              "fontSize": 15,
              "marginBottom": 12,
            }
          }
        >
          Max es de raza labrador
        </Text>
        <View
          style={
            [
              {
                "alignSelf": "flex-start",
                "borderRadius": 20,
                "marginBottom": 8,
                "paddingHorizontal": 12,
                "paddingVertical": 6,
              },
              {
                "backgroundColor": "#d4edda",
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#212529",
                "fontSize": 14,
                "fontWeight": "500",
              }
            }
          >
            Disponible para adopción
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
